src/persist-txn/src/operator.rs

```diff
@@ -527,20 +528,29 @@ where
 
 /// The process global [`TxnsRead`] that any operator can communicate with.
 #[derive(Default, Debug, Clone)]
-pub struct TxnsContext<T: Clone> {
-    read: Arc<tokio::sync::OnceCell<TxnsRead<T>>>,
+pub struct TxnsContext {
+    read: Arc<tokio::sync::OnceCell<Box<dyn Any + Send + Sync>>>,
```

Comment:
Why do we need `dyn Any + Send + Sync` here instead of something like `TxnsRead<T>` or `TxnsRead<dyn Any + Send + Sync>`?

Reply:
It used to be `TxnsRead<T>` but then that `T` ends up spreading all the way up through this plumbing in a way that I suspected would be objectionable (compute, like persist, doesn't refer to `mz_repr::Timestamp` directly, instead abstracting over it). And then `TxnsRead<dyn Any + Send + Sync>` doesn't work like you'd want it to: `Box<dyn Any>` has this `downcase_ref` method, but you can't do that generally.